mmv1/products/discoveryengine/ChatEngine.yaml

```diff
@@ -0,0 +1,176 @@
+    path: 'error'
+    message: 'message'
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engineId'
+    description: |
+      The ID to use for chat engine.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collectionId.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'industryVertical'
+    description: |
+      Industry Vertical
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      displayName
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    min_size: 1
+    description: |
+      Data Store IDs, Data Stores cannot be removed once added.
+      Removing a Data Store from Chat Engine will result in error.
+      To resolve error, first use `terraform refresh` command to correct state,
+      then revert the code change to ensure graceful teardown.
+    required: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Enum
+    name: 'solutionType'
+    description: |
+      Solution type for Chat Engine is SOLUTION_TYPE_CHAT
+    values:
+      - :SOLUTION_TYPE_CHAT
+    default_value: :SOLUTION_TYPE_CHAT
+    immutable: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'chatEngineConfig'
+    description: |
+      Configurations for a chat Engine.
+    required: true
+    immutable: true
+    ignore_read: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'agentCreationConfig'
+        description: |
+          Configurations for a chat Engine.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'business'
+            description: |
+              Business Name
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'defaultLanguageCode'
+            description: |
+              Default Language Code
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'timeZone'
+            description: |
+              time zone
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'location'
+            description: |
+              location
+            required: true
```

Comment:
location is not required

Reply:
We need this field in URL and the path
e.g. 

https://{{**location**}}-discoveryengine.googleapis.com/v1/projects/{{project}}/locations/{{**location**}}/collections/{{collection_id}}/engines/{{engine_id}}

Reply:
Oh this is the agentCreationConfig.location, which is different from the top-level location (line 74). This one is optional.

Reply:
Ah good catch! I uas confused it with the engine location parameter. ok I'll update this.